source/IrcDotNet/StandardIrcClient.cs

```diff
@@ -341,20 +363,36 @@ private void SendCompleted(object sender, SocketAsyncEventArgs e)
             }
         }
 
-        private void ReceiveAsync()
+        private async void ReceiveAsync()
         {
             // Read data received from socket to buffer asynchronously.
-            var receiveEventArgs = new SocketAsyncEventArgs();
-            Debug.Assert(receiveStream.Buffer.Length - (int) receiveStream.WritePosition > 0);
-            receiveEventArgs.SetBuffer(receiveStream.Buffer, (int) receiveStream.WritePosition,
-                receiveStream.Buffer.Length - (int) receiveStream.WritePosition);
-            receiveEventArgs.Completed += ReceiveCompleted;
-
-            if (!socket.ReceiveAsync(receiveEventArgs))
-                ReceiveCompleted(socket, receiveEventArgs);
+            var receiveEventArgs = new SyntheticSocketAsyncEventArgs(); //can't set BytesTransferred on the SocketAsyncEventArgs so use synthetic instead
```

Comment:
Not necessarily a bad idea, but a) deserves a better name, b) a comment like this better belongs on the class itself.

Reply:
Any suggestions for a better name?

Reply:
Possibly something boring like `SocketAsyncEventArgs2` — it seems to be a common convention at least. Or perhaps even keep the same name, and just use a type alias to distinguish between the two classes... but I'm less sure about the sensibility of that.